test/suites/integration/bsp/volunteer.test.ts

```diff
@@ -194,118 +191,124 @@ describeBspNet("Single BSP Volunteering", ({ before, createBspApi, it, createUse
-    const bucketName = "nothingmuch-3";
+describeBspNet(
+  "Multiple BSPs volunteer ",
+  { keepAlive: false },
+  ({ before, createBspApi, createUserApi, it }) => {
+    let userApi: EnrichedBspApi;
+    let bspApi: EnrichedBspApi;
 
-    const newBucketEventEvent = await userApi.createBucket(bucketName);
-    const newBucketEventDataBlob =
-      userApi.events.fileSystem.NewBucket.is(newBucketEventEvent) && newBucketEventEvent.data;
-
-    if (!newBucketEventDataBlob) {
-      throw new Error("Event doesn't match Type");
-    }
+    before(async () => {
+      userApi = await createUserApi();
+      bspApi = await createBspApi();
+    });
 
-    const txs = [];
-    for (let i = 0; i < source.length; i++) {
-      const { fingerprint, file_size, location } =
-        await userApi.rpc.storagehubclient.loadFileInStorage(
-          source[i],
-          destination[i],
-          userApi.shConsts.NODE_INFOS.user.AddressId,
-          newBucketEventDataBlob.bucketId
+    it("bsp volunteers multiple files properly", async () => {
+      const source = ["res/whatsup.jpg", "res/adolphus.jpg", "res/smile.jpg"];
+      const destination = ["test/whatsup.jpg", "test/adolphus.jpg", "test/smile.jpg"];
+      const bucketName = "nothingmuch-3";
+
+      const newBucketEventEvent = await userApi.createBucket(bucketName);
+      const newBucketEventDataBlob =
+        userApi.events.fileSystem.NewBucket.is(newBucketEventEvent) && newBucketEventEvent.data;
+
+      if (!newBucketEventDataBlob) {
+        throw new Error("Event doesn't match Type");
+      }
+
+      const txs = [];
+      for (let i = 0; i < source.length; i++) {
+        const { fingerprint, file_size, location } =
+          await userApi.rpc.storagehubclient.loadFileInStorage(
+            source[i],
+            destination[i],
+            userApi.shConsts.NODE_INFOS.user.AddressId,
+            newBucketEventDataBlob.bucketId
+          );
+
+        txs.push(
+          userApi.tx.fileSystem.issueStorageRequest(
+            newBucketEventDataBlob.bucketId,
+            location,
+            fingerprint,
+            file_size,
+            userApi.shConsts.DUMMY_MSP_ID,
+            [userApi.shConsts.NODE_INFOS.user.expectedPeerId]
+          )
         );
-
-      txs.push(
-        userApi.tx.fileSystem.issueStorageRequest(
-          newBucketEventDataBlob.bucketId,
-          location,
-          fingerprint,
-          file_size,
-          userApi.shConsts.DUMMY_MSP_ID,
-          [userApi.shConsts.NODE_INFOS.user.expectedPeerId]
-        )
+      }
+
+      await userApi.sealBlock(txs, shUser);
+
+      //There should be pending extrinsics for all files from BSP (volunteer)
+      await userApi.assert.extrinsicPresent({
+        module: "fileSystem",
+        method: "bspVolunteer",
+        checkTxPool: true,
+        assertLength: source.length
+      });
+
+      await userApi.sealBlock();
+
+      await sleep(5000); // wait for the bsp to download the files
```

Comment:
i dont know why we need this sleep() but inconsistent behaviour with the nodes happen without it (the fileKeys attribute of the event only shows 1 key)